x-pack/plugins/index_management/public/application/components/mappings_editor/__jest__/client_integration/mappings_editor.test.tsx

```diff
@@ -166,12 +205,183 @@ describe('<MappingsEditor />', () => {
+     * as it is the only place where it is consumed by the mappings editor.
+     */
+    const defaultMappings: any = {
+      dynamic: true,
+      numeric_detection: false,
+      date_detection: true,
+      properties: {
+        title: { type: 'text' },
+        address: {
+          type: 'object',
+          properties: {
+            street: { type: 'text' },
+            city: { type: 'text' },
+          },
+        },
+      },
+      dynamic_templates: [{ initial: 'value' }],
+      _source: {
+        enabled: true,
+        includes: ['field1', 'field2'],
+        excludes: ['field3'],
+      },
+      _meta: {
+        some: 'metaData',
+      },
+      _routing: {
+        required: false,
+      },
+    };
+
+    let testBed: MappingsEditorTestBed;
+
+    beforeEach(async () => {
+      testBed = await setup({ value: defaultMappings, onChange: onChangeHandler });
+    });
+
+    test('props.value => should prepopulate the editor data', async () => {
+      const {
+        actions: { selectTab, getJsonEditorValue, getComboBoxValue, getToggleValue },
+        find,
+      } = testBed;
+
+      /**
+       * Mapped fields
+       */
+      expect(find('fieldsListItem').length).toEqual(Object.keys(defaultMappings.properties).length);
+
+      const fields = find('fieldsListItem.fieldName').map(item => item.text());
+      expect(fields).toEqual(Object.keys(defaultMappings.properties).sort());
+
+      /**
+       * Dynamic templates
+       */
+      await act(async () => {
+        await selectTab('templates');
+      });
+
+      const templatesValue = getJsonEditorValue('dynamicTemplatesEditor');
+      expect(templatesValue).toEqual(defaultMappings.dynamic_templates);
+
+      /**
+       * Advanced settings
+       */
+      await act(async () => {
+        await selectTab('advanced');
+      });
+
+      const isDynamicMappingsEnabled = getToggleValue(
+        'advancedConfiguration.dynamicMappingsToggle.input'
+      );
+      expect(isDynamicMappingsEnabled).toBe(defaultMappings.dynamic);
+
+      const isNumericDetectionEnabled = getToggleValue(
+        'advancedConfiguration.numericDetection.input'
+      );
+      expect(isNumericDetectionEnabled).toBe(defaultMappings.numeric_detection);
+
+      expect(getComboBoxValue('sourceField.includesField')).toEqual(
+        defaultMappings._source.includes
+      );
+      expect(getComboBoxValue('sourceField.excludesField')).toEqual(
+        defaultMappings._source.excludes
+      );
+
+      const metaFieldValue = getJsonEditorValue('advancedConfiguration.metaField');
+      expect(metaFieldValue).toEqual(defaultMappings._meta);
+
+      const isRoutingRequired = getToggleValue('advancedConfiguration.routingRequiredToggle.input');
+      expect(isRoutingRequired).toBe(defaultMappings._routing.required);
+    });
+
+    test('props.onChange() => should forward the changes to the consumer component', async () => {
+      let updatedMappings = { ...defaultMappings };
```

Comment:
Do we need this line? Can we declare `updatedMappings` on line 324 like this?

```js
      const newField = { name: getRandomString(), type: 'text' };
      const updatedMappings = {
        ...defaultMappings,
      };
      updatedMappings.properties[newField.name] = { type: newField.type };
```

If we can split up these assertions into multiple `test` calls then we won't need to continue overwriting the same `updatedMappings` variable.

Reply:
Like I said above the focus has been made in limiting the number of `test()` 
 and `setup()` calls, and this is a small tradeoff.

In this case, I prefer to make a copy first at the beginning of the test and then only add the modifications to the object when it should have changed.